### PR TITLE
Set deepvariant as the sole variant caller

### PR DESCRIPTION
**Pull request by ramprasadn**
### This PR adds | fixes:

- Moves baserecalibration to haplotypecaller chain
- Turns-off haplotypecaller and deeptrio
- Uses deepvariant as the sole variant caller
- Glnexus uses unfiltered config by default for both WGS and WES samples. 
- Recipes that were once downstream of baserecalibration will now work with outputs from markduplicates

### How to test:

- Automatic and continuous test suite

### Expected outcome:
- [ ] Installation, unit and integration test suite pass

### Review:
- [x] Code review
- [x] Tests pass

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes. If applicable record manual test results in PR header
- [x] **MINOR** - when you add functionality in a backwards compatible manner. If applicable record manual test results in PR header
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Review by jemten — APPROVED** [review]
Looks good. You should also change the deeptrio and deepvariant recipes to not explicitly say markduplicates when you get the infiles. You should be able to use `$recipe_name` instead now

File: lib/MIP/Recipes/Pipeline/Analyse_rd_dna.pm [hunk 1]
```diff
@@ -220,11 +220,13 @@ sub parse_rd_dna {
     );
 
     ## Check that the supplied gatk sample map file paths exists
-    check_gatk_sample_map_paths(
-        {
-            sample_map_path => $active_parameter_href->{gatk_genotypegvcfs_ref_gvcf},
-        }
-    );
+    if ( $parameter_href->{gatk_genotypegvcfs}{default} ) {
```
[the remark this answers is not in the document]
Comment (ramprasadn) [reply lib/MIP/Recipes/Pipeline/Analyse_rd_dna.pm]: Good point! Fixed 👍🏻

File: lib/MIP/Recipes/Pipeline/Analyse_rd_dna.pm [hunk 2]
```diff
@@ -221,11 +221,13 @@ sub parse_rd_dna {
     );
 
     ## Check that the supplied gatk sample map file paths exists
-    check_gatk_sample_map_paths(
-        {
-            sample_map_path => $active_parameter_href->{gatk_genotypegvcfs_ref_gvcf},
-        }
-    );
+    if ( $active_parameter_href->{gatk_genotypegvcfs} ) {
```
Comment (henrikstranneheim) [inline lib/MIP/Recipes/Pipeline/Analyse_rd_dna.pm:224]: I think you can put this logic in the sub and do :
```
return if (not $active_parameter_href->{gatk_genotypegvcfs} );
```
early

**Review by henrikstranneheim — APPROVED** [review]
Nice! Maybe you should change the title of the PR and provide some  more scope in the PR header.

**Comment by henrikstranneheim** [comment]
Also need some more coverage before merging

File: lib/MIP/Recipes/Pipeline/Analyse_rd_dna.pm (hunk 2 above)
Comment (ramprasadn) [reply lib/MIP/Recipes/Pipeline/Analyse_rd_dna.pm]: Fixed! 👍🏻

**Pull request merged**